extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -314,7 +314,9 @@ public void build(HibernateValidatorRecorder recorder, RecorderContext recorderC
         // JAX-RS methods are handled differently by the transformer so those need to be gathered here.
         // Note: The focus only on methods is basically an incomplete solution, since there could also be
         // class-level JAX-RS annotations but currently the transformer only looks at methods.
-        Map<DotName, Set<SimpleMethodSignatureKey>> jaxRsMethods = gatherJaxRsMethods(additionalJaxRsResourceMethodAnnotations,
+        Set<DotName> additional = new HashSet<>();
+        additionalJaxRsResourceMethodAnnotations.forEach((s) -> additional.addAll(s.getAnnotationClasses()));
```

Comment:
I think you forgot to update the import of `AdditionalJaxRsResourceMethodAnnotationsBuildItem` from the old SPI (it should import the one from the new SPI).

Maybe you could also remove that class from the old SPI, just to be sure it doesn't get used again by mistake?